internal/ingest/ingest.go

```diff
@@ -140,9 +120,19 @@ func NewIngester(cfg config.Ingest, h host.Host, idxr indexer.Interface, reg *re
 		closeWorkers:            make(chan struct{}),
 	}
 
+	// Construct a selector that recursively looks for nodes with field
+	// "PreviousID" as per Advertisement schema. Note that the entries within
+	// an advertisement are synced separately, triggered by storage hook. This
+	// allows checking if a chain of chunks already exists before syncing it.
+	ssb := builder.NewSelectorSpecBuilder(basicnode.Prototype.Any)
+	adSequence := ssb.ExploreFields(
+		func(efsb builder.ExploreFieldsSpecBuilder) {
+			efsb.Insert("PreviousID", ssb.ExploreRecursiveEdge())
+		}).Node()
```

Comment:
these could be top level variables, not re-defined each time the function runs. They are are also available in the segmented_sync.go file, so should try not to shadow.